src/plugin.js

```diff
@@ -110,60 +137,63 @@ videojs.HlsSourceHandler = function(mode) {
           tech.trigger('loadstart');
         }, 1);
       }
-      tech.hls = new videojs.HlsHandler(tech, {
-        source: source,
-        mode: mode
+      tech.hls = new HlsHandler(tech, {
+        source,
+        mode
       });
       tech.hls.src(source.src);
       return tech.hls;
     },
-    canPlayType: function(type) {
-      return videojs.HlsSourceHandler.canPlayType(type);
+    canPlayType(type) {
+      return HlsSourceHandler.canPlayType(type);
     }
   };
 };
 
-videojs.HlsSourceHandler.canPlayType = function(type) {
-  var mpegurlRE = /^application\/(?:x-|vnd\.apple\.)mpegurl/i;
+HlsSourceHandler.canPlayType = function(type) {
+  let mpegurlRE = /^application\/(?:x-|vnd\.apple\.)mpegurl/i;
 
   // favor native HLS support if it's available
-  if (videojs.Hls.supportsNativeHls) {
+  if (Hls.supportsNativeHls()) {
     return false;
   }
   return mpegurlRE.test(type);
 };
 
 // register source handlers with the appropriate techs
-if (videojs.MediaSource.supportsNativeMediaSources()) {
-  videojs.getComponent('Html5').registerSourceHandler(videojs.HlsSourceHandler('html5'));
+if (MediaSource.prototype.supportsNativeMediaSources()) {
```

Comment:
These type functions should be static functions so that they are at the constructor level and not in the prototype.